### PR TITLE
OCPBUGS-18772: resolv-prepender: avoid pulling baremetalRuntimeCfgImage again 

### PR DESCRIPTION
**Pull request by vrutkovs**
Avoid pulling baremetalRuntimeCfgImage again if it already exists on the node

`pull_baremetal_runtime_cfg_image` function should avoid  pull attempt if this image already present on the node. This ensures DNS prepender can work when mirror/quay image is temporarily unreachable

**- What I did**
Updated resolv-prepender.sh to make sure that pull is not running if the image already exists on the node 

**- How to verify it**
Check networkmanager logs - it should have `NM resolv-prepender: Starting download of baremetal runtime cfg image` just once (in upgrade case you may see it twice) for on-premise jobs, i.e.:
```
wget -O - https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3907/pull-ci-openshift-machine-config-operator-master-e2e-vsphere/1701162972396654592/artifacts/e2e-vsphere/gather-extra/artifacts/nodes/ci-op-8s25i4px-6b38f-9262w-master-0/journal | zgrep "Starting download"    
--2023-09-11 13:27:38--  https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3907/pull-ci-openshift-machine-config-operator-master-e2e-vsphere/1701162972396654592/artifacts/e2e-vsphere/gather-extra/artifacts/nodes/ci-op-8s25i4px-6b38f-9262w-master-0/journal
Resolving gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com (gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com)... 52.205.222.12, 52.23.49.187
Connecting to gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com (gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com)|52.205.222.12|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/gzip]
Saving to: ‘STDOUT’

-                                                   [  <=>                                                                                                  ] 243.70K   589KB/s               Sep 11 09:35:54.580706 ci-op-8s25i4px-6b38f-9262w-master-0 resolv-prepender.sh[2139]: NM resolv-prepender: Starting download of baremetal runtime cfg image
-                                                   [   <=>                                                                                                 ] 712.49K  1.26MB/s    in 0.6s    

2023-09-11 13:27:39 (1.26 MB/s) - written to stdout [729591]
```

**- Description for the changelog**

resolve-prepender no longer needs reachable image source

### CODE REVIEW

**Comment by vrutkovs** [comment]
/hold

vsphere IPI is fine, but metal and assisted tests are broken

**Comment by vrutkovs** [comment]
/hold cancel

Passing now

**Review by yuqi-zhang — APPROVED** [review]
Seems fine logically. Since this doesn't seem urgent, will wait for kube rebase to stabalize before merging

**Comment by sergiordlr** [comment]
Double checked with the e2e-metal-ip installation.

```
$ curl -sk https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/3907/pull-ci-openshift-machine-config-operator-master-e2e-metal-ipi/1701233381662003200/artifacts/e2e-metal-ipi/gather-extra/artifacts/nodes/master-0/journal | gunzip | grep "NM resolv-prepender: Starting download of baremetal runtime cfg image"
Sep 11 14:32:45.010790 master-0 resolv-prepender.sh[2001]: NM resolv-prepender: Starting download of baremetal runtime cfg image
```

We can add the qe-approved label

/label qe-approved

**Comment by sinnykumari** [comment]
LGTM
/cc @cybertron PTAL

**Comment by cybertron** [comment]
/lgtm

/cc @mkowalski 
Maybe you're already aware, but it looks like this is the change we discussed recently mitigate the weird empty resolv.conf issues.

**Pull request merged**
